src/mam/mod_mam_muc.erl

```diff
@@ -116,6 +120,23 @@
 %% ----------------------------------------------------------------------
 %% API
 
+-spec get_personal_data(jid:user(), jid:server()) ->
+    [{gdpr:data_group(), gdpr:schema(), gdpr:entries()}].
+get_personal_data(Username, Server) ->
+    Schema = ["id", "message"],
+    Entries = lists:flatmap(
+        fun(B) ->
+            try B:get_mam_muc_gdpr_data(Username, Server) of
+                {ok, GdprData} ->
+                    GdprData;
+                _ -> []
+            catch
+                _:_ ->
```

Comment:
Error should be reported.
Also, just returning empty data instead of an error is very confusing for users.